src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_MediaPlayerElement.cs

```diff
@@ -11,38 +13,247 @@
 
 using _MediaPlayer = Windows.Media.Playback.MediaPlayer; // alias to avoid same name root namespace from ios/macos
 
-namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls;
+
+[TestClass]
+[RunsOnUIThread]
+public partial class Given_MediaPlayerElement
 {
-	[TestClass]
-	[RunsOnUIThread]
-	public partial class Given_MediaPlayerElement
+	private static readonly Uri TestVideoUrl = new Uri("https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/720/Big_Buck_Bunny_720_10s_5MB.mp4");
+
+	[TestMethod]
+	public async void When_NotAutoPlay_Source()
 	{
-		private static readonly Uri TestVideoUrl = new Uri("https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/720/Big_Buck_Bunny_720_10s_5MB.mp4");
+		Test_Setup();
+		var sut = new MediaPlayerElement()
+		{
+			AutoPlay = false,
+			Source = MediaSource.CreateFromUri(TestVideoUrl),
+			Name = "When_NotAutoPlay_Source",
+			Width = 100,
+		};
+		WindowHelper.WindowContent = sut;
+		await WindowHelper.WaitForLoaded(sut);
+
+		// PlaybackState should transition out of Opening state when the video is ready to play.
+		await WindowHelper.WaitFor(
+			condition: () => sut.MediaPlayer?.PlaybackSession?.PlaybackState == Windows.Media.Playback.MediaPlaybackState.Paused,
+			timeoutMS: 5000,
+			message: "Timeout waiting for the media player to enter Paused state."
+		);
+		WindowHelper.WindowContent = null;
```

Comment:
maybe you can move this to a method with [TestCleanup]?

Reply:
Removed.